cpp/community_detection_module/algorithm/label_propagation.cpp

```diff
@@ -0,0 +1,291 @@
+#include <cmath>
+#include <mg_graph.hpp>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include "label_propagation.hpp"
+
+namespace LabelRankT {
+
+///@brief Checks if set is subset of another set.
+///
+///@tparam T -- set element data type
+///
+///@param set1 -- subset candidate
+///@param set2 -- set against which the check is performed
+///
+///@return -- whether set1 is subset of set2
+template <class T>
+bool is_subset(std::vector<T> const& set1, std::vector<T> const& set2) {
+  for (const auto& element : set1) {
+    if (std::find(set2.begin(), set2.end(), element) == set2.end())
+      return false;
+  }
+
+  return true;
+}
+
+double LabelRankT::get_weight(std::uint64_t node_i_id,
+                              std::uint64_t node_j_id) {
+  // WORK IN PROGRESS
+
+  return DEFAULT_WEIGHT;
+}
+
+void LabelRankT::set_structures(std::uint64_t node_id) {
+  times_updated[node_id] = 0;
+  std::unordered_map<std::uint64_t, double> node_label_Ps;
+
+  double sum_w_i = w_selfloop;
+  for (const auto node_j : graph->Neighbours(node_id))
+    sum_w_i += get_weight(node_j.node_id, node_id);
```

Comment:
Use curly brackets even when there is only one line of code in the for-loop. It makes code easier to read.

Reply:
Thanks for the tip!